static/app/views/issueList/customViewsHeader.tsx

```diff
@@ -257,20 +259,49 @@ function CustomViewsIssueListHeaderTabsContent({
               tab.id = view.id;
             }
           });
-          navigate({
-            ...location,
-            query: {
-              ...queryParams,
-              viewId: tab.id,
+          navigate(
+            {
+              ...location,
+              query: {
+                ...queryParams,
+                viewId: tab.id,
+              },
             },
-          });
+            {replace: true}
+          );
```

Comment:
Unrelated change: this makes it so when we replace a temporary viewId with the persistent Id, nothing is added to the navigation history, which is the expected behavior.